### PR TITLE
Bug 1156204 - Add support for PKCS#12 files

### PR DESCRIPTION
**Pull request by crh0716**
https://bugzilla.mozilla.org/show_bug.cgi?id=1156204

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3f1aaa0c0fe4947986f36ea9af503472d18ad92b

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=26d4ea8fc83dce22b41b86ee5b4a74e3655a5285

File: apps/settings/js/panels/wifi_select_certificate_file/panel.js [hunk 1]
```diff
@@ -71,38 +71,87 @@ define(function(require) {
...
+          } else {
+            password = result;
+            return this._requestNickName(certificateName);
+          }
+        }).then((result) => {
+          nickname = result;
+          return new Promise((resolve, reject) => {
+            var req = wifiManager.importCert(file, password, nickname);
+            req.onsuccess = resolve;
+            req.reject = onerror;
```
Comment (chuck-lee) [inline apps/settings/js/panels/wifi_select_certificate_file/panel.js]: I think here should be |req.onerror = reject;|

Comment (crh0716) [inline apps/settings/js/panels/wifi_select_certificate_file/panel.js]: !! good catch

Comment (chuck-lee) [inline apps/settings/js/panels/wifi_select_certificate_file/panel.js]: I use following form to bypass error message from domrequest.onerror() to promise.reject() to test Bug 1163031.
```
req.onerror = function (e) {
  reject(e.target.error.name);
}
```

**Pull request closed without merging**